### ISSUE TITLE
plotXMLAttributes.py

### ISSUE THREAD

**Issue by shiyu022** [eclipse/sumo#13451]
Add your issue description here.

*If possible, upload an example that shows your problem.*

**SUMO-version:**

**operating system:**

**Comment by shiyu022** [comment]
when I use these two commands
![截屏2023-06-19 16 14 58](https://github.com/eclipse/sumo/assets/101698390/661050ba-a893-4745-b0d5-f4a2cf1f95c2)
![截屏2023-06-19 16 14 58](https://github.com/eclipse/sumo/assets/101698390/9517fd35-aaaf-4e57-8d20-420dac21ba07)
unicodeEncodeError will be presented in my terminal, and I don't know how to fix it
![截屏2023-06-19 16 16 50](https://github.com/eclipse/sumo/assets/101698390/702eb1d3-a976-42d0-81fb-f5d1fb9a3b53)
so I am preparing my paper, learning visualization method of sumo, but things are not going well and I need your help.
Thank you!

**Comment by behrisch** [comment]
Can you share your input file?

**Comment by shiyu022** [comment]
> Can you share your input file?
![截屏2023-06-19 19 44 08](https://github.com/eclipse/sumo/assets/101698390/511140bf-d3f9-494f-9e8d-6d22b1f726ca)

![截屏2023-06-19 19 43 55](https://github.com/eclipse/sumo/assets/101698390/8d88b9f4-e425-4141-bad0-ec47ccfe7762)

**Comment by shiyu022** [comment]
> could also be a python2/python3 issue. What python version are you using?

![截屏2023-06-19 19 46 34](https://github.com/eclipse/sumo/assets/101698390/0d09655c-4279-4f29-9e02-7c313a2f421b)
can you teach me how to do? thanks

**Comment by namdre** [comment]
use python3

**Comment by shiyu022** [comment]
> use python3
![截屏2023-06-19 19 56 43](https://github.com/eclipse/sumo/assets/101698390/efc0da89-2569-4185-965d-a389c130dca0)

what to do next and how to fix it?

**Comment by namdre** [comment]
call `python3 plotXMLAttributes ...` where `...` are all the arguments from the example.

**Comment by shiyu022** [comment]
> call `python3 plotXMLAttributes ...` where `...` are all the arguments from the example.

one more question
plot_net_speeds.py reads a road network and plots it using the allowed speeds to color the edges.
I was wondering if there is any tool that could reads the average speed of a road section in a network and color the edges?

**Comment by namdre** [comment]
for this we have https://sumo.dlr.de/docs/Tools/Visualization.html#plot_net_dumppy
see also https://sumo.dlr.de/docs/Simulation/Output/Lane-_or_Edge-based_Traffic_Measures.html#visualization_plotting

**Comment by shiyu022** [comment]
I have a question about how to alter the name of legend. My command line is as follows:
python3 plotXMLAttributes.py summary.xml summary1.xml -x time -y meanSpeed --xlabel "Time" --ylabel "Meanspeed" --title "Comparison of Meanspeed" --xlim 0,3500 -o plot-running.png --legend

I want to revise the name of legend. what should I add to my command line? thank you @namdre @namdre 
![图](https://github.com/eclipse/sumo/assets/101698390/3fbf3f2d-e01a-4643-be08-7a3de990baaf)

**Comment by namdre** [comment]
@shiyu022 you can add the desired LEGEND_NAME into the summary.xml file by setting attribute `id="LEGEND_NAME"` in the root element (`<summary>`) of the respective input file. In the absence of an id attribute, any part of the file name that differs between the two input files will be added after a hash mark (hence the current `#1` for file *summary1.xml*. Thus you can also influence the legend by renaming the files.

**Comment by shiyu022** [comment]
> @shiyu022 you can add the desired LEGEND_NAME into the summary.xml file by setting attribute `id="LEGEND_NAME"` in the root element (`<summary>`) of the respective input file. In the absence of an id attribute, any part of the file name that differs between the two input files will be added after a hash mark (hence the current `#1` for file _summary1.xml_. Thus you can also influence the legend by renaming the files.

Yeah, I do change the name of the legend by renaming the two files, but I don't want the '0#'. is there any method to delete it?
![截屏2023-07-03 15 27 32](https://github.com/eclipse/sumo/assets/101698390/1c451826-4cd6-4021-84d3-a8bdeaeb707c)

**Comment by namdre** [comment]
put the `id` attribute into the root element of the summary.xml files (then the names do not matter).

**Comment by namdre** [comment]
also in the latest development version, the full filename will be used without `#` if you set option **--idattr @NONE**  (#134534)

**Comment by shiyu022** [comment]
> Also in the latest development version you don't need to modify the xml file contents. Instead, the full filename will be used without `#` if you set option **--idattr @NONE** (#13534)

my sumo version is 1.17.0

**Comment by namdre** [comment]
The latest release is 1.18.0 (but that doesn't contain the fix). You can download the latest development version with the fix tomorrow at https://sumo.dlr.de/docs/Downloads.php#nightly_snapshots

**Comment by shiyu022** [comment]
> The latest release is 1.18.0 (but that doesn't contain the fix). You can download the latest development version with the fix tomorrow at https://sumo.dlr.de/docs/Downloads.php#nightly_snapshots

ok, thank you for teaching me so much about SUMO. if there are some problems that I don't know, I will ask you for help😂

**Comment by shiyu022** [comment]
How to export HD images? @namdre  I already set the dpi, but the effect is not obvious
![截屏2023-07-04 15 14 38](https://github.com/eclipse/sumo/assets/101698390/36596a0c-af67-4a03-ba4a-f13d3eeb3fd0)

**Comment by namdre** [comment]
- either export as vector graphic (pdf, ps, svg) instead of png
- or set higher dpi (you should notice that the file size increases)
